src/builds/builds.service.ts

```diff
@@ -29,6 +29,19 @@ export class BuildsService {
     });
   }
 
+  async deleteOldBuilds(projectId: string, keepBuilds: number) {
+    const [total, buildList] = await Promise.all([
+      this.prismaService.build.count({ where: { projectId } }),
+      this.prismaService.build.findMany({
+        where: { projectId },
+        orderBy: { createdAt: 'desc' },
+      }),
+    ]);
+    for (let index = (keepBuilds - 1); index < total; index++) {
+      this.eventsGateway.buildDeleted(buildList[index]);
```

Comment:
didn't find where are the builds actually removed?

Reply:
Sorry, actually I was trying out with fronted code and forgot to include one line, fixed now.